Tests.Unit.Presentation.Web/Services/ReferenceServiceTest.cs

```diff
@@ -432,61 +432,103 @@ public void AddReference_Does_Not_Save_Root_If_Add_Fails()
+            var externalReferencePropertiesList = CreateExternalReferenceProperties(false).ToList();
+            var expectedMaster = externalReferencePropertiesList.RandomItem();
             expectedMaster.MasterReference = true;
 
+            var entity = root.Object.FromNullable();
             var masterReference = CreateExternalReference(expectedMaster);
             ExpectMasterReference(root, masterReference);
-            ExpectRootDeleteAndAdd(root, externalReferences, new List<ExternalReference>());
+            ExpectRootDeleteAndAdd(root, externalReferencePropertiesList, new List<ExternalReference>());
             ExpectAllowModifyReturns(root.Object, true);
             ExpectTransactionToBeSet();
-            ExpectGetRootEntityReturns(rootId, rootType, root.Object.FromNullable());
+            ExpectGetRootEntityReturns(rootId, rootType, entity);
 
             //Act
-            var result = _sut.BatchUpdateExternalReferences(rootType, rootId, externalReferences);
+            var result = _sut.UpdateExternalReferences(rootType, rootId, externalReferencePropertiesList);
 
             //Assert
             Assert.True(result.IsNone);
             _dbTransaction.Verify(x => x.Commit());
+            foreach (var externalReference in externalReferencePropertiesList)
+            {
+                root.Verify(x => x.AddExternalReference
+                    (
+                        It.Is<ExternalReference>
+                        (
+                            reference => reference.Title == externalReference.Title
+                                         && reference.ExternalReferenceId == externalReference.DocumentId
+                                         && reference.URL == externalReference.Url)
+                    ),
+                    Times.Once
+                );
+            }
         }
 
         [Fact]
-        public void Can_BatchUpdateExternalReferences_With_Previous_References()
+        public void Can_UpdateExternalReferences()
         {
             //Arrange
             var rootType = A<ReferenceRootType>();
             var rootId = A<int>();
+            var externalReference = new ExternalReference
+            {
+                Uuid = Guid.NewGuid()
+            };
+            var externalReferenceList = new List<ExternalReference>
+            {
+                externalReference
+            };
             var root = new Mock<IEntityWithExternalReferences>();
+            root.SetupGet(x => x.ExternalReferences).Returns(externalReferenceList);
+            var rootEntity = root.Object;
+
             Configure(f => f.Inject(false)); //Make sure no master is added when faking the inputs
-            var externalReferences = Many<UpdatedExternalReferenceProperties>().ToList();
-            var expectedMaster = externalReferences.RandomItem();
-            expectedMaster.MasterReference = true;
+            var externalReferencePropertiesList = new List<UpdatedExternalReferenceProperties>()
```

Comment:
do we validate the case where the we see all of the effects:
- A reference is updated
- A reference is added
- An existing reference is removed